### PR TITLE
[MOBILE] Fix Firebase Crashlytics issues

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
#### Summary of Changes (What does this PR do?)

- Fixes the following firebase crashlytics issues.
- [x] [Unmodifiable list on favorite places](https://console.firebase.google.com/u/0/project/airqo-250220/crashlytics/app/android:com.airqo.app/issues/8c254f3d3197ee78ddb94e3086ac28fa?time=last-seven-days&sessionEventKey=6387426C039F000E59F19CEED827FF14_1750219567680244185)
- [x] [Unmodifiable list on analytics](https://console.firebase.google.com/u/0/project/airqo-250220/crashlytics/app/android:com.airqo.app/issues/785979444dbb735ba1b438536f9d246e?time=last-seven-days&sessionEventKey=63874E32026F00045317D43437D1DB3F_1750233075178393040)
- [x] [Unmodifiable list on KYA](https://console.firebase.google.com/u/0/project/airqo-250220/crashlytics/app/android:com.airqo.app/issues/b9620c568ab6b399771d15b7817195a0?time=last-seven-days&sessionEventKey=63871880020100131F1B89B501EF0BAD_1750173475513844573)

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Checkout this branch
- Run the app and it should be working as expected.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [MOB-152](https://airqoteam.atlassian.net/browse/MOB-152) (If exists)

### CODE REVIEW

**Review by NoahNsimbe — APPROVED** [review]
LGTM

File: mobile/lib/blocs/account/account_bloc.dart [hunk 1]
```diff
@@ -38,7 +38,7 @@ class AccountBloc extends Bloc<AccountEvent, AccountState> {
   ) async {
     emit(state.copyWith(blocStatus: BlocStatus.updatingData));
 
-    final favouritePlaces = state.favouritePlaces;
+    List<FavouritePlace> favouritePlaces = state.favouritePlaces;
```
Comment (BenjaminSsempala) [inline mobile/lib/blocs/account/account_bloc.dart:41]: Making this non final such that the variable can be modified

File: mobile/lib/blocs/account/account_bloc.dart [hunk 2]
```diff
@@ -267,7 +267,7 @@ class AccountBloc extends Bloc<AccountEvent, AccountState> {
 
     emit(state.copyWith(blocStatus: BlocStatus.updatingData));
 
-    final kya = state.kya;
+    List<Kya> kya = state.kya;
```
Comment (BenjaminSsempala) [inline mobile/lib/blocs/account/account_bloc.dart:270]: Making this non final such that the variable can be modified

File: mobile/lib/screens/analytics/analytics_view.dart [hunk 3]
```diff
@@ -24,7 +24,7 @@ class AnalyticsView extends StatelessWidget {
           context.read<AccountBloc>().add(const RefreshAnalytics());
         }
 
-        final analytics = state.analytics.sortByDateTime();
+        List<Analytics> analytics = state.analytics.sortByDateTime();
```
Comment (BenjaminSsempala) [inline mobile/lib/screens/analytics/analytics_view.dart:27]: Making this non final such that the variable can be modified

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @BenjaminSsempala

**Pull request merged**